src/Chapter3.hs

```diff
@@ -910,6 +988,21 @@ Implement instances of "Append" for the following types:
 class Append a where
     append :: a -> a -> a
 
+newtype Gold = Gold Int
+instance Append Gold where
+    append :: Gold -> Gold -> Gold
+    append (Gold g1) (Gold g2) = Gold (g1+g2)
+
+instance Append [a] where
+    append :: [a] -> [a] -> [a]
+    append lst1 lst2 = lst1 ++ lst2
+
+instance (Append a) => Append (Maybe a) where
+    append :: Maybe a -> Maybe a -> Maybe a
+    append Nothing Nothing = Nothing
+    append Nothing (Just x) = Just x
+    append (Just x) Nothing = Just x
```

Comment:
This is an absolutely correct solution!

However, we can notice that these 3 cases could be reduced to 2!
Let's see what we can unite in here.

If some of the arguments are `Nothing`, you actually always return the second one! So you need to inspect only two cases: `Nothing` in the first position, and `Nothing` on the second one 🙂